src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -66,31 +73,245 @@ public override string Description
+        {
+            _isDeprecatedMode = true;
+            return base.Using(comparer);
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied boolean-returning delegate.
+        /// </summary>
+        /// <param name="comparer">The supplied boolean-returning delegate to use.</param>
+        [Obsolete(ObsoleteMessage)]
+        public new CollectionItemsEqualConstraint Using<T>(Func<T, T, bool> comparer)
+        {
+            _isDeprecatedMode = true;
+            return base.Using(comparer);
+        }
+
+        #endregion
+
+        private static MethodInfo GetContainsKeyMethod(object keyedItemContainer)
+        {
+            if (keyedItemContainer == null) throw new ArgumentNullException(nameof(keyedItemContainer));
+            var instanceType = keyedItemContainer.GetType();
+
+            var method = FindContainsKeyMethod(instanceType)
+                         ?? instanceType
+                            .GetInterfaces()
+                            .Concat(GetBaseTypes(instanceType))
+                            .Select(FindContainsKeyMethod)
+                            .FirstOrDefault(m => m != null);
+
+            return method;
+        }
+
+        private static MethodInfo FindContainsKeyMethod(Type type)
+        {
+            var methods = type.GetMethods(BindingFlags.Instance | BindingFlags.Public);
+            var method = methods.FirstOrDefault(m =>
+                m.ReturnType == typeof(bool)
+                && m.Name == "ContainsKey"
+                && !m.IsGenericMethod
+                && m.GetParameters().Length == 1);
+
+            if (method == null && type.GetTypeInfo().IsGenericType)
+            {
+                var definition = type.GetGenericTypeDefinition();
+                var tKeyGenericArg = definition.GetGenericArguments().FirstOrDefault(typeArg => typeArg.Name == "TKey");
+
+                if (tKeyGenericArg != null)
+                {
+                    method = definition
+                             .GetMethods(BindingFlags.Instance | BindingFlags.Public)
+                             .FirstOrDefault(m =>
+                                 m.ReturnType == typeof(bool)
+                                 && m.Name == "Contains"
+                                 && !m.IsGenericMethod
+                                 && m.GetParameters().Length == 1
+                                 && m.GetParameters()[0].ParameterType == tKeyGenericArg);
+
+                    if (method != null)
+                    {
+#if NETSTANDARD1_4
+                        method = methods.Single(m => Matched(method, m));
+#else
+                        method = methods.Single(m => m.MetadataToken == method.MetadataToken);
+#endif
+                    }
+                }
+            }
+
+            return method;
+        }
+
+#if NETSTANDARD1_4
+        
+        //
+        // This is for missing MetadataToken in NetStandard 1.4, matched by method signature
+        //
```

Comment:
We don't have spacing with empty `//` lines anywhere else, so this sticks out. Dunno if it matters at all.